packages/client/src/components/sidebar-right.js

```diff
@@ -175,6 +175,13 @@ const component = (state, emit) => {
               <div
                 class='Sidebar-list-item UserList-user ${user.online ? 'is-online' : 'is-offline'}'
                 title='${user.username}${user.flair ? ` {${user.flair}}` : ''} (${user.online ? 'Online' : 'Offline'})'
+                onclick=${() => {
+                  const textarea = document.querySelector('.MessageEditor-textarea')
+
+                  if (textarea) {
+                    textarea.value += '@' + user.username
```

Comment:
This should insert a space before the `@` if (the textarea is not empty && the last character is not a space / at the start of the line).

Also, this should totally insert at wherever the cursor is positioned in the textarea, but don't worry about that now!! It's super a pain to implement.